flashlight/lib/CMakeLists.txt

```diff
@@ -56,6 +56,8 @@ if (FL_LIBRARIES_USE_CUDA)
   include(${CMAKE_MODULE_PATH}/CUDAUtils.cmake)
 
   set_cuda_cxx_compile_flags()
+  # Required when using cmake 3.12+ not using anymore CUDA_NVCC_FLAGS:
+  target_compile_options(fl-libraries PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--std=c++14>)
   set_cuda_arch_nvcc_flags()
```

Comment:
Have you tested this change with CMake < 3.12 to make sure this compile definition isn't added? Adding the c++ flag twice might cause nvcc to fail...

Reply:
yes, I ve just tried and builds correctly with cmake 3.10 : only 1 --std=c++14 :
[ 16%] Building CUDA object CMakeFiles/fl-libraries.dir/flashlight/lib/sequence/criterion/cuda/ViterbiPath.cu.o
/usr/local/cuda/bin/nvcc  -DFL_LIBRARIES_USE_CUDA -DFL_LIBRARIES_USE_KENLM -DKENLM_MAX_ORDER=6 -I/home/wtambellini/repos/flashlight -I/usr/include/openblas -I/home/wtambellini/repos/kenlm -isystem=/usr/include -I/home/wtambellini/repos/flashlight/debug/CUB/src/CUB -I/usr/local/cuda/include  -g   --std=c++14 -x cu -c /home/wtambellini/repos/flashlight/flashlight/lib/sequence/criterion/cuda/ViterbiPath.cu -o CMakeFiles/fl-libraries.dir/flashlight/lib/sequence/criterion/cuda/ViterbiPath.cu.o
Is nt your CI jobs anyway building with cmake 3.10 ?

Reply:
Yep, they are building with CMake 3.10. I think the issue I've had before with this was pre-CMake 3.10, so let's commit this and hope nothing breaks :)